### ISSUE TITLE
Perfetto scripts require an arm64 python interpreter on Apple M1

### ISSUE THREAD

**Issue by primiano** [google/perfetto#294]
This bug summarizes a class of issues user a reporting on Apple arm-based laptops.

### Symptom

Following the quick-start instructions like [this](https://perfetto.dev/docs/quickstart/trace-analysis) leads to an error like this:

```
./trace_processor trace.perfetto-trace --http
This executable requires a cpu that supports SSE4.2 and AVX2.
Rebuild with enable_perfetto_x64_cpu_opt=false (ebx=10800, ecx=298220f).
```

### Root cause

Background:
- The quickstart instructions prompt the user to download and execute scripts from get.perfetto.dev
- That URL serves precisely the ToT version of python the scripts [`tools/traceconv`](https://github.com/google/perfetto/blob/master/tools/traceconv), [`tools/trace_processor`](https://github.com/google/perfetto/blob/master/tools/trace_processor) and the like
- Those Python scripts are a thin wrapper which downloads the actual native prebuilts (the ones we attach to each GitHub release like [this](https://github.com/google/perfetto/releases/tag/v25.0)) and execute them. Essentially they take care of downloading the right binary for the right architecture.
- We detect the architecture using python's `platform.machine()`. In general the architecture will match the architecture of the python interpreter that is being used to execute them.

Further twists:
- We have both native x86_64 and arm64 binaries for MacOS
- Our x86_64 prebuilds assume SSE4.2 and AVX2. This is because that makes `trace_processor_shell` noticeably faster.
- Rosetta (The x86_64-on-arm64 emulator that MacOS ships on M1) does NOT emulate SSE4.2

In short: the problem arises when running these scripts on an arm64 Mac machine using an x86_64 version of python which runs under Rosetta. This causes the scripts to download and spawn the x86_64 version of our prebuilts rather than the arm64 ones.

### Fix

Make sure that the default python3 interpreter installed is a native arm64 binary, not a x86_64 that runs under rosetta.

`python3 -c 'import platform; print(platform.machine())' should return arm64 not x86_64

See also:
 - https://laict.medium.com/install-python-on-macos-11-m1-apple-silicon-using-pyenv-12e0729427a9
 - https://stackoverflow.com/questions/65559632/how-can-i-run-python-3-9-1-natively-on-m1-mac
 - https://developer.apple.com/forums/thread/695963
 - https://docs.python.org/3/whatsnew/3.9.html
